napari/layers/utils/layer_utils.py

```diff
@@ -684,58 +684,201 @@ def features_to_pandas_dataframe(features: Any) -> pd.DataFrame:
+
+    @property
+    def values(self) -> pd.DataFrame:
+        """The feature values table."""
+        return self._values
+
+    def set_values(self, values, *, num_data=None):
+        """Sets the feature values table."""
+        self._values = _validate_features(values, num_data=num_data)
+        self._defaults = self._make_defaults()
+
+    def _make_defaults(self) -> pd.DataFrame:
+        """Makes the default values table from the feature values."""
+        return pd.DataFrame(
+            {
+                name: _get_default_column(column)
+                for name, column in self._values.items()
+            },
+            index=range(1),
+            copy=True,
         )
-    else:
-        features = _validate_features(features, num_data=num_data)
-    defaults = pd.DataFrame(
-        {
-            name: _get_default_column(column)
-            for name, column in features.items()
-        },
-        index=range(1),
-        copy=True,
-    )
-    return features, defaults
+
+    @property
+    def defaults(self) -> pd.DataFrame:
+        """The default values one-row table."""
+        return self._defaults
+
+    def properties(self) -> Dict[str, np.ndarray]:
+        """Converts this to a deprecated properties dictionary.
+
+        This will reference the features data when possible, but in general the
+        returned dictionary mary contain copies of those data.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            The properties dictionary equivalent to the given features.
+        """
+        return _features_to_properties(self._values)
+
+    def choices(self) -> Dict[str, np.ndarray]:
+        """Converts this to a deprecated property choices dictionary.
+
+        Only categorical features will have corresponding entries in the dictionary.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            The property choices dictionary equivalent to this.
+        """
+        return {
+            name: series.dtype.categories.to_numpy()
+            for name, series in self._values.items()
+            if isinstance(series.dtype, pd.CategoricalDtype)
+        }
+
+    def currents(self) -> Dict[str, np.ndarray]:
+        """Converts the defaults table to a deprecated current properties dictionary."""
+        return _features_to_properties(self._defaults)
+
+    def set_currents(
+        self,
+        currents: Dict[str, np.ndarray],
+        *,
+        update_indices: Optional[List[int]] = None,
+    ):
+        """Sets the default values using the deprecated current properties dictionary.
+
+        May also update some of the feature values to be equal to the new default values.
+        """
+        currents = coerce_current_properties(currents)
+        self._defaults = _validate_features(currents, num_data=1)
+        if update_indices is not None:
+            for k in self._defaults:
+                self._values[k][update_indices] = self._defaults[k][0]
```

Comment:
I'm confused as to if this could make defaults out of sync with values. I haven't tried to completely get the logic though, just noting my confusion.

Reply:
By out of sync, do you mean that `self._defaults` may contain different keys to `self._values`, so this could error?

If so, I agree and this is one of the reasons I'm planning to deprecate this behavior. I could add an explicit validation step to check that the keys are the same, but as this behavior is soon to be deprecated, I think it's OK to let the error be raised from elsewhere (e.g. pandas).